README

			Graph-ReadWrite
			      2.02

This is a collection of perl classes for reading and writing
directed graphs in a variety of file formats. The graphs are
represented in Perl using Jarkko Hietaniemi's Graph classes.
Version 2+ of this distribution works with version 0.5 and
later of the Graph distribution.

There are two base classes:

	Graph::Reader
	Base class for classes which read a graph
	file and create an instance of the Graph class.

	Graph::Writer
	Base class for classes which take an instance of the
	Graph class and write it out in a specific file format.

This distribution also defines a simple XML format for storing
directed graphs, defined to meet the needs of the Graph class.
A reader and writer are provided for this:

	Graph::Reader::XML
	Graph::Writer::XML

Writers are also included for Dot, VCG, and daVinci,
two packages for displaying / rendering graphs.
Dot and VCG are freely available; daVinci is freely available
at the moment, but that is going to change soon.

	Graph::Writer::Dot
	Graph::Writer::VCG
	Graph::Writer::daVinci

A reader for Dot files has been contributed by Mark Hillebrand:

	Graph::Reader::Dot

A reader and writer for HTK lattices are also included:

	Graph::Reader::HTK
	Graph::Writer::HTK

These last two will probably only be of interest to people working
with HTK, or tools which use the same format.

Dependencies:
	Graph		Jarkko's Graph classes
	XML::Parser	used by Graph::Reader::XML
	XML::Writer	used by Graph::Writer::XML
	IO::File

To install these modules, you should just have to run the following:

	% perl Makefile.PL
	% make
	% make test
	% make install

This distribution contains an early version of a test-suite;
you may get a warning about an undefined value from one of the tests.

Bugs, suggestions, and new readers and writers will be gratefully received.

Neil Bowers <neil@bowers.com>


Changes
Revision history for perl distribution Graph-ReadWrite

2.08 2015-10-30 NEILB
    - Switched to Dist::Zilla
    - Updated the github repo URL after changing my github username
    - Removed a spurious =back from Graph::ReadWrite.
      Thanks to Lucas Kanashiro for RT#106063

2.07 2015-02-21 NEILB
    - Made mentions in SEE ALSO be links. MANWAR++ for starting this.
    - Added Graph::ReadWrite module, so there's a module that matches
      the dist name.
    - The NAME field in Makefile.PL was set to dist name (Graph-ReadWrite)
      instead of module name (Graph::ReadWrite).

2.06 2014-01-24

    - Graph::Write::daVinci had a different version number from
      all the other modules.

2.05 2014-01-22

    - the daVinci writer would sometimes fail due to the random
      hash order change in Perl 5.18
    - Specified min perl version as 5.6.0
    - Added ExtUtils::MakeMaker to CONFIGURE_REQUIRES
    - Changed Makefile.PL to metaspec 2

2.04 2013-07-27

    - Added license type to Metadata
    - Ensured all releases in this file have a date, as per CPAN::Changes::Spec

2.03 2012-02-14

    - added ABSTRACT, AUTHOR and github repository info to Makefile.PL
    - added parent, Carp, and File::Compare as pre-reqs

2.02 2012-02-12

    - Pulled out the Parse::Yapp::Driver code that was embedded in the
      Dot reader. PAUSE didn't like me uploading code for a module I
      don't have permission for. Now added as a pre-req.

2.01 2012-02-12

    - Fixed RT bug #39792: trying to call method set_graph_attribute()
      on Graph::Reader::XML, which should have been set_attribute().
    - changed calls to File::Compare::compare() to hopefully cope with
      EOL differences between Windows and Unix. Reported by Philip Potter,
      and fix noted by Max
    - Ensured all modules were use'ing strict and warnings. Fixed warnings :-)
    - Cleaned up formatting of all modules
    - renamed ChangeLog to Changes and reformatted as per CPAN::Changes::Spec

2.00 2005-01-02

	- made the changes needed to work with version 0.5 of Graph classes
	- patch to Graph::Reader::Dot from Mark Hillebrand.
	- fixed warnings "Using a hash as a reference is deprecated"
	- added support for new Dot attributes to Graph::Reader::Dot
	- node ids are now quoted in Graph::Writer::Dot,
	  which means you can safely have space, ':', '-', and other
	  interesting characters in node ids.

1.07 2002-07-09

	- Added XML::Parser and XML::Writer as dependencies in Makefile.PL.
	  This was prompted by looking at testers.cpan.org, which has
	  this distribution as failing.

1.06 2002-03-20

	- Graph::Writer::VCG - common attributes (label, color, and
	  text color) weren't supported on edges.
	  Bug and fix reported by Alexandre Cherif.

1.05 2002-01-20

	- Added a basic test-suite.
	- Commented out debugging output in the HTK Writer. Sorry Phil!

1.04 2001-11-11

	- Added Graph::Reader::Dot, a reader for the Dot format
	  contributed by Mark Hillebrand.

	- Fixed a bug in Graph::Reader::XML, found by Mark Hillebrand.
	  The reader would only cope with a single <attribute> element
	  inside each graph, node or edge element.

	- Changed contact email address to neil at bowers dot com

1.03 2001-04-25

	- Added reader and writer classes for HTK lattice files.

1.02 2001-03-18

	- Fixed a bug in the XML writer module - the way it was using
	  XML::Writer caused STDOUT to be closed after you've written
	  a graph. I consider it a bug in XML::Writer, but I was able
	  to change the way Graph::Writer::XML uses that module.

	  Basically it just means that we construct a new instance
	  of XML::Writer every time we write out a graph. For most
	  uses, this won't be a problem, but you'd notice a hit if
	  you were creating the writer and then writing a lot of graphs.

	  This was noticed by Mike Hayward <hayward@loup.net>

1.01 2001-03-12

	- Added support for URL attribute on nodes in Dot.
	- Added support for almost all attributes understood by VCG.
	- Added a rough cut at a Writer class for daVinci: Graph::Writer::daVinci

1.00 2001-02

	- Initial release - has baseclass for Reader and Writer classes.
	- Reader and Writer for simple XML format.
	- Writers for Dot and VCG formats.



Makefile.PL
#
# Makefile.PL for Graph-ReadWrite
#

use ExtUtils::MakeMaker;

my $mm_ver = $ExtUtils::MakeMaker::VERSION;
if ($mm_ver =~ /_/) { # dev version
    $mm_ver = eval $mm_ver;
    die $@ if $@;
}

&WriteMakefile(
    NAME         => 'Graph::ReadWrite',
    DISTNAME     => 'Graph-ReadWrite',
    ABSTRACT     => 'Classes for reading and writing directed graphs',
    AUTHOR       => 'Neil Bowers <neil@bowers.com>',
    VERSION      => '2.07',
    LICENSE      => 'perl',

    ($mm_ver <= 6.45
        ? ()
        : (META_MERGE => {
            'meta-spec' => { version => 2 },
            resources => {
                bugtracker  => 'http://rt.cpan.org/Public/Dist/Display.html?Name=Graph-ReadWrite',
                repository  => {
                    type => 'git',
                    web  => 'https://github.com/neilbowers/Graph-ReadWrite',
                    url  => 'git://github.com/neilbowers/Graph-ReadWrite.git',
                },
            },
          })
    ),

    PREREQ_PM    => {
                     'IO::File'            => 0,
                     'Graph'               => 0.5,
                     'XML::Parser'         => 0,
                     'XML::Writer'         => 0,
                     'Parse::Yapp::Driver' => 0,
                     'parent'              => 0,
                     'Carp'                => 0,
                     'File::Compare'       => 0,
                    },

    ($mm_ver >= 6.48
        ? (MIN_PERL_VERSION => 5.006)
        : ()
    ),

    ($mm_ver >= 6.52
        ? (CONFIGURE_REQUIRES => {
                'ExtUtils::MakeMaker' => 6.30,
          })
        : ()
    ),

    dist         => {COMPRESS => 'gzip', SUFFIX => 'gz'},
);

